Tests/PubNubContractTest/Steps/Objects/PubNubObjectsChannelMetadataContractTestSteps.swift

```diff
@@ -0,0 +1,220 @@
+//  PubNub Real-time Cloud-Hosted Push API and Push Notification Client Frameworks
+//  Copyright © 2022 PubNub Inc.
+//  https://www.pubnub.com/
+//  https://www.pubnub.com/terms
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Cucumberish
+import Foundation
+import PubNub
+import XCTest
+
+public class PubNubObjectsChannelMetadataContractTestSteps: PubNubContractTestCase {
+  var channelMetadata: PubNubChannelMetadata?
+  
+  public override func handleBeforeHook() {
+    channelMetadata = nil
+    super.handleBeforeHook()
+  }
+  
+  public override func setup() {
+    startCucumberHookEventsListening()
+    
+    Given("^the id for '(.*)' channel$") { args, _ in
+      guard args?.count == 1, let channelName = args?.first?.lowercased() else {
+        XCTAssert(false, "Channel name not specified.")
+        return
+      }
+      
+      guard let channelMetadata = self.channelMetadata(with: channelName) else {
+        XCTAssert(false, "Channel file not parsed.")
+        return
+      }
+      
+      XCTAssertNotNil(channelMetadata.metadataId)
+      self.channelMetadata = PubNubChannelMetadataBase(metadataId: channelMetadata.metadataId)
+    }
+    
+    Given("^the data for '(.*)' channel$") { args, _ in
+      guard args?.count == 1, let channelName = args?.first?.lowercased() else {
+        XCTAssert(false, "Channel name not specified.")
+        return
+      }
+      
+      guard let channelMetadata = self.channelMetadata(with: channelName) else {
+        XCTAssert(false, "Channel file not parsed.")
+        return
+      }
+      
+      self.channelMetadata = channelMetadata
+    }
+    
+    When("^I get the channel metadata(.*)$") { args, _ in
+      let fetchChannelMetadataExpect = self.expectation(description: "Fetch channel metadata response")
+      var includeCustom = false
+      
+      guard let channelMetadataId = self.channelMetadata?.metadataId else { return }
```

Comment:
Do we want to make an assertion here?

Reply:
@CraigLn this one goes after [this](https://github.com/pubnub/swift/pull/118/files/20ecc010f24ad9c6e3b11f67be3d37728c66fd10#diff-4f7451017693fb9dfd82099613a2239b52b80f6f235bf4f3dc13d8692343d37bR44) step and it already have check on `metadataId`